controllers/util/version.go

```diff
@@ -0,0 +1,104 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type kernelSupportVersion struct {
+	// Latest is the kernel version in which the issue was fixed upstream.
+	// Any version greater than this is considered to have the fix.
+	Latest string
+
+	// Backports contains base kernel version prefixes that received backported fixes.
+	// Matching logic: match the first 4 version segments, and the 5th segment should be greater
+	// than or equal to the backported entry to consider the fix as present.
+	Backports []string
+}
+
+var (
+	CephxKeyRotaionKernelSupportVersion = kernelSupportVersion{
+		Latest:    "5.14.0-570.22.1",
```

Comment:
The comment for `Latest` refers to _upstream_, this does not look like an upstream version. Does `semver.Make()` work with a version like `6.16.0`?